### ISSUE TITLE
FckEditor - Malforming HTML when loading

### ISSUE THREAD

**Issue by Noodleyman** [cubecart/v6#684]
Fckeditor is parsing HTML source code incorrectly from the DB. 

I am working on moving some of the template files to a document for easier editing, however ran into a problem because fckeditor is not playing nice.

Here is an example:
take the HTML code from the admin template file orders.print.php.

Create a new document and paste the HTML code into it in SOURCE mode.

Save the document.

check the code in the database. It is correct. 

When you edit the document and put it back into SOURCE mode the code loaded has been malformed resulting in an unusable document if saved again.

There is a disconenct between the source being loaded from the DB and the source being used in the document to be edited.

**Comment by bhsmither** [comment]
Some of the template files are complete web pages. `orders.print.php` is one of them. The CKEditor has a special mode (http://docs.ckeditor.com/#!/guide/dev_fullpage) that allows for complete page editing. That special mode is currently **only** enabled when editing the email/newsletter templates (not the email content templates).

If you change documents.index.php, line 84, this part:
`class="textbox fck"` to `class="textbox fck fck-full"`
then the editor understands that you want a complete HTML page and will not disturb HTML section tags and sections. **However**, the editor will add whatever is necessary to make the document a **complete** HTML document if the tags and sections are not there.

This will make invalid a regular site document.

I would try to create a plugin to give me a specifically configured editor to edit complete HTML documents. Maybe test for `<!DOCTYPE` in the content to auto-switch.

**Comment by Noodleyman** [comment]
Morning Brian,

Thanks for sharing this info. very useful to know.

I did some testing prior to opening the ticket, and I was able to reproduce the same behaviour with e-mail templates so I suspect this mode isn't enabled by default for them.

I didn't test newsletter template yet.

I will play around with it and I can enable the full mode as required on a per document basis via hooks. will report back my findings when I've had chance to revisit this item
